submissions/sofiichuk/oop-exercise/index.js

```diff
@@ -0,0 +1,75 @@
+class Animal {
+    constructor(name, gender, saying, legs) {
+        this.name = name;
+        this.gender = gender;
+        this.saying = saying;
+        this.legs = legs;
+    }
+
+    declareAnimal() {
+        return [this.name, this.gender, this.saying, this.legs].join(";");
+    }
+}
+
+class Human extends Animal {
+    constructor(name, gender, saying, legs = 2, arms = 2) {
+        super(name, gender, saying, legs);
+        this.arms = arms;
+    }
+
+    declareGenus() {
+        return [this.constructor.name, super.declareAnimal(), this.arms].join(";");
```

Comment:
Check what is the output of `this.constructor.name` when it is used in the root class.
You will possibly reduce code repetition.
BTW, there was no requirement to have classes for men and women, just for species.
I think dogs and cat may feel discriminated, as they do not have privilege of gender-related classes like `HeDog/SheDog` and `HeCat/SheCat`.

Reply:
hi.
i will fix what you're asking for considering the code, but there's another point. 
you ask to set classNames/classHierarchy properly. 
no way, the exercise is initially broken. 
there's a general trouble with class naming regarding this task. 
i've gotta philosophical education (and also i'm a pedant) (and it spoils everything).
i need man-women classes for the purpose of some symmetry.
due to aristotelis, we define anything by naming its genus+sigificant_feature (walnut is a tree with nuts). 
the base class should be the god, but let's skip it, it's a joke. 
definitely, the base-js-class is an animal. but if we dig deeper into the taxonomy - levels of human & cats-dogs (canivals) are not equal. parets-childs has a bunch of levels between also. and so on, and so on. what i'm trying to say: ANY approach to naming in this certain task is falthy. the only thing we can do - minimizing stupidity of simpliefied approach

Reply:
i mean we write a code not only for the machines, but also for other people. 
(the text above is a bit messy, but nevertheless)

Reply:
removed redundant classes. and also species printing is now defined by the root class